sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -769,16 +769,21 @@ class CodegenContext {
       foldFunctions: Seq[String] => String = _.mkString("", ";\n", ";")): String = {
     val blocks = new ArrayBuffer[String]()
     val blockBuilder = new StringBuilder()
+    val maxLines = SQLConf.get.maxCodegenLinesPerFunction
```

Comment:
This flag does not take an effect at runtime.

Reply:
What do you mean? Could you elaborate this review comment?

Reply:
If you set the conf at runtime, can you get the values from `SQLConf.get.maxCodegenLinesPerFunction`?

Reply:
I see. I had another interpretation that "this value may not change performance".
Let me check this while I did it like other flags.

Reply:
Got it. Depends on calling context, it may take an effect or not. Should we pass `SQLConf` to this method?

Reply:
I made this option effective by executing `SparkEnv.get.conf`

Reply:
@gatorsmile Since to make it configurable [takes long time](https://github.com/apache/spark/pull/19449#discussion_r143385878), can we do it using hard-coded parameter?
Even in this case, this PR makes better since the estimation does not include characters for comment.

Reply:
@kiszk You know, I am just afraid new regression could be introduced due to this change. Sorry for the delay. I really do not have a better solution. I kind of agree on your original solution. Just exclude the characters for comment. At least, it becomes better and take a less risk to hit a regression.

Reply:
@gatorsmile I understand your concern about the possibility of new performance regression. I will use the original threshold (max characters) as hard-coded value.